### PR TITLE
Update clang-format

### PR DESCRIPTION
**Pull request by ddn0**
We don't plan to actively sync with ISS/Galois, so we don't need to
follow their code formatting. The number of active developers is small
enough that we can afford a couple more tweaks the code formatting
before it gets prohibitively expensive.

Compared to the previous style, there is considerably less horizontal
whitespace alignment, which should be helpful for avoiding merge
conflicts.

The formatting style is based off the Google style, with the following
notable exceptions:
- in places where the Google style allows some freedom, prefer a
  canonical output (e.g., AllowShortLoopsOnASingleLine: false,
  AlwaysBreakAfterReturnType: TopLevelDefinitions, DeriveXYZ: false)
- include ordering was adjusted to allow third-party inputs to be
  included by `#include <foo/bar.h>` rather than #include "foo/bar.h"

### CODE REVIEW

File: libava/manager/manager.cpp [hunk 1]
```diff
@@ -115,9 +120,9 @@ class DaemonServiceClient {
   DaemonServiceClient(std::shared_ptr<grpc::Channel> channel)
       : channel_(channel), stub_(DaemonService::NewStub(channel)) {}
 
-  std::vector<ServerAddress> SpawnWorker(unsigned count,
-                                         std::vector<std::string>& uuid,
-                                         std::vector<uint64_t>& gpu_mem) {
+  std::vector<ServerAddress> SpawnWorker(
+      unsigned count, std::vector<std::string>& uuid,
+      std::vector<uint64_t>& gpu_mem) {
     /* Build message. */
```
Comment (witchel) [inline libava/manager/manager.cpp:126]: I prefer seeing the first parameter on the call line and aligned parameters below, but I can adjust.

**Comment by witchel** [comment]
> * in places where the Google style allows some freedom, prefer a
>   canonical output (e.g., AllowShortLoopsOnASingleLine: false,
>   AlwaysBreakAfterReturnType: TopLevelDefinitions, DeriveXYZ: false)

I'm a big fan of more strict defaults and no short loops on a single line.

> * include ordering was adjusted to allow third-party inputs to be
>   included by `#include <foo/bar.h>` rather than #include "foo/bar.h"

Any help with include files is appreciated.  I know I'm supposed to order and group them, but I can never quite remember the proper order and grouping...

**Comment by ddn0** [comment]
> Any help with include files is appreciated. I know I'm supposed to order and group them, but I can never quite remember the proper order and grouping...

With this change, `clang-format` will do it for you, so it should be automatic :D

**Comment by tylershunt** [comment]
It's got everything I want and nothing offensive to me thanks for taking the time to do this!

**Comment by scober** [comment]
Thanks for doing this!

**Comment by danielmawhirter** [comment]
No objection here, definitely happy to have automatic include ordering!

**Review by roshandathathri — APPROVED** [review]
I have no objections and there was sufficient time for others to voice their objections. Let's merge this ASAP!

Thanks @ddn0 !

**Pull request merged**